include/sisl/sobject/sobject.hpp

```diff
@@ -0,0 +1,116 @@
+/*********************************************************************************
+ * Modifications Copyright 2017-2019 eBay Inc.
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ *********************************************************************************/
+#pragma once
+
+#include <string>
+#include <functional>
+#include <map>
+#include <set>
+#include <memory>
+#include <shared_mutex>
+#include <vector>
+#include <nlohmann/json.hpp>
+
+#include <sisl/logging/logging.h>
+
+namespace sisl {
+
+struct sobject_id {
+    std::string type;
+    std::string name;
+    bool empty() const { return type.empty() && name.empty(); }
+};
+
+typedef struct status_request {
+    nlohmann::json json;
+    bool do_recurse{false};
+    int verbose_level = 0;
+    std::string obj_type;
+    std::string obj_name;
+    std::vector< std::string > obj_path;
+    int batch_size = 10;
+    std::string next_cursor;
+} status_request;
+
+typedef struct status_response {
+    nlohmann::json json;
+} status_response;
+
+using status_callback_type = std::function< status_response(const status_request&) >;
+class sobject;
+using sobject_ptr = std::shared_ptr< sobject >;
+
+[[maybe_unused]] static bool operator<(const sobject_id& id1, const sobject_id& id2) {
```

Comment:
nitpick: Is it possible to move these three operator to be inside struct sobject_id?